File: delivery-service/delivery/db/src/main/java/org/example/delivery/db/storemenu/StoreMenuRepository.java
package org.example.delivery.db.storemenu;

import org.example.delivery.db.storemenu.enums.StoreMenuStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface StoreMenuRepository extends JpaRepository<StoreMenuEntity, Long> {

    /**
     * SELECT * FROM store_menu WHERE id = ? AND status = ? ORDER BY id DESC LIMIT 1;
     */
    Optional<StoreMenuEntity> findFirstByIdAndStatusOrderByIdDesc(Long id, StoreMenuStatus status);

    /**
     * SELECT * FROM store_menu WHERE store_id = ? AND status = ? ORDER BY sequence DESC;
     */
    List<StoreMenuEntity> findAllByStoreIdAndStatusOrderBySequenceDesc(Long storeId, StoreMenuStatus status);
}

File: delivery/db/src/main/kotlin/org/example/delivery/db/storemenu/StoreMenuRepository.kt
package org.example.delivery.db.storemenu

import org.example.delivery.db.storemenu.enums.StoreMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreMenuRepository : JpaRepository<StoreMenuEntity, Long> {
    /**
     * SELECT * FROM store_menu WHERE id = ? AND status = ? ORDER BY id DESC LIMIT 1;
     */
    fun findFirstByIdAndStatusOrderByIdDesc(id: Long?, status: StoreMenuStatus?): StoreMenuEntity?

    /**
     * SELECT * FROM store_menu WHERE store_id = ? AND status = ? ORDER BY sequence DESC;
     */
    fun findAllByStoreIdAndStatusOrderBySequenceDesc(storeId: Long?, status: StoreMenuStatus?): List<StoreMenuEntity>
}
